src/coverlet.core/Instrumentation/Instrumenter.cs

```diff
@@ -393,6 +424,11 @@ private void InstrumentType(TypeDefinition type)
                         customAttributes = customAttributes.Union(prop.CustomAttributes);
                 }
 
+                if (IsMethodOfCompilerGeneratedClassOfAsyncStateMachineToBeExcluded(method))
+                {
+                    continue;
+                }
+
                 ordinal++;
```

Comment:
I think this instruction must be moved before the if

Reply:
Done can you take a look?